File: conciertosBack/api/src/main/java/com/rubenazo/buscaConciertos/adapters/in/dto/ArtistDto.java
package com.rubenazo.buscaConciertos.adapters.in.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.rubenazo.buscaConciertos.domain.Artist;

public record ArtistDto(
    String id,
    String name,
    String genre,
    @JsonProperty("image_url") String imageUrl,
    String website,
    String description,
    @JsonProperty("source_url") String sourceUrl
) {
    public static ArtistDto from(Artist artist) {
        return new ArtistDto(
            artist.id(),
            artist.name(),
            artist.genre(),
            artist.imageUrl(),
            artist.website(),
            artist.description(),
            artist.sourceUrl()
        );
    }
}

File: conciertosFront/shared/src/commonMain/kotlin/org/rubenazo/conciertosfront/core/data/remote/dto/ArtistDto.kt
package org.rubenazo.conciertosfront.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ArtistDto(
    @SerialName("id") val id: String,
    @SerialName("name") val name: String,
    @SerialName("genre") val genre: String?,
    @SerialName("image_url") val imageUrl: String?,
    @SerialName("website") val website: String?,
    @SerialName("description") val description: String?,
    @SerialName("source_url") val sourceUrl: String?
)
